pos_loyalty_card_fixed_expiration_date/models/loyalty_card.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 (APSL-Nagarro) - Antoni Marroig
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date, timedelta
+
+from odoo import api, models
+
+
+class LoyaltyCard(models.Model):
+    _inherit = "loyalty.card"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        # If the expiration date is not set, we set it to
+        # the current date + the fixed duration
+        if vals_list:
+            if not vals_list[0].get("expiration_date", False):
```

Comment:
This doesn't seem right.

`vals_list` here can be a list containing completely different values. You shouldn't rely only on the first one, for all the records.

Instead you can do something like this:

```py
for record, vals in zip(res, vals_list):
    ...
```

This works because create will return the records on the same order than the provided values that created them.

---

But moreover, I wonder if it wouldn't be a better idea to simply make the `expiration_date` a computed field with `store=True`, `readonly=False`, `precompute=True`, and `@api.depends('program_id')`, instead of overriding `create`.